test/unit/controllers/declaration/ExporterEoriNumberControllerSpec.scala

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers.declaration
+
+import controllers.declaration.ExporterEoriNumberController
+import forms.common.YesNoAnswer.YesNoAnswers
+import forms.common.{Address, Eori}
+import forms.declaration.EntityDetails
+import forms.declaration.exporter.{ExporterDetails, ExporterEoriNumber}
+import models.{DeclarationType, Mode}
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, times, verify, when}
+import org.scalatest.OptionValues
+import play.api.data.Form
+import play.api.libs.json.Json
+import play.api.mvc.{AnyContentAsEmpty, Request}
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import unit.base.ControllerSpec
+import views.html.declaration.exporter_eori_number
+
+class ExporterEoriNumberControllerSpec extends ControllerSpec with OptionValues {
+
+  val mockExporterEoriNumberPage = mock[exporter_eori_number]
+
+  val controller = new ExporterEoriNumberController(
+    mockAuthAction,
+    mockJourneyAction,
+    navigator,
+    stubMessagesControllerComponents(),
+    mockExporterEoriNumberPage,
+    mockExportsCacheService
+  )(ec)
+
+  def checkViewInteractions(noOfInvocations: Int = 1): Unit =
+    verify(mockExporterEoriNumberPage, times(noOfInvocations)).apply(any(), any())(any(), any())
+
+  def theResponseForm: Form[ExporterEoriNumber] = {
+    val captor = ArgumentCaptor.forClass(classOf[Form[ExporterEoriNumber]])
+    verify(mockExporterEoriNumberPage).apply(any(), captor.capture())(any(), any())
+    captor.getValue
+  }
+
+  override def getFormForDisplayRequest(request: Request[AnyContentAsEmpty.type]): Form[_] = {
+    withNewCaching(aDeclaration(withType(DeclarationType.CLEARANCE)))
+    await(controller.displayPage(Mode.Normal)(request))
+    theResponseForm
+  }
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    authorizedUser()
+    when(mockExporterEoriNumberPage.apply(any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    super.afterEach()
+    reset(mockExporterEoriNumberPage)
+  }
+
+  "should return a 200 (OK)" when {
+    onJourney(DeclarationType.STANDARD, DeclarationType.SUPPLEMENTARY, DeclarationType.OCCASIONAL, DeclarationType.SIMPLIFIED) { request =>
```

Comment:
Nit: It is easier to read when there is no `DeclarationType` before each journey type ;-)

Reply:
Also, shouldn't this clause be run against ALL journey types? On line 210 starts some tests for Clearance and returning 303, but there is nothing for Clearance and 200 response.

Reply:
Done.